### PR TITLE
MINIFICPP-1779 Verify multiple C2 commands in HB response

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1779

---------------------------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by szaszm — APPROVED** [review]

File: extensions/http-curl/tests/C2MultipleCommandsTest.cpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      is_odd_ack = true;
+    }
+    if (root.IsObject() && root.HasMember("operationId")) {
+      ack_auditor_.addAck(root["operationId"].GetString());
+    }
+  }
+
+ private:
+  AckAuditor& ack_auditor_;
+  bool is_odd_ack = true;
```
Comment (adamdebreceni) [inline extensions/http-curl/tests/C2MultipleCommandsTest.cpp:70]: checking validity based on parity seems strange, shouldn't we branch based on the operation id?

Comment (lordgamez) [reply extensions/http-curl/tests/C2MultipleCommandsTest.cpp:70]: We would like to guarantee that the C2 commands are processed in their order in the request, that's why we are checking this way.

Comment (adamdebreceni) [reply extensions/http-curl/tests/C2MultipleCommandsTest.cpp:70]: in that case, we could maintain a queue of ack verifier functions, that would make it seamless to add new operations

Comment (lordgamez) [reply extensions/http-curl/tests/C2MultipleCommandsTest.cpp]: Updated in 1ade7ca6ec6c6feb7498f4b3be7d86f4e78b1fdb

File: extensions/http-curl/tests/C2MultipleCommandsTest.cpp [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      assert(false);
+    }
+
+    if (next_verifier_index_ >= ack_verifiers_.size()) {
+      ack_verifiers_[0](root);
+      next_verifier_index_ = 1;
+    } else {
+      ack_verifiers_[next_verifier_index_](root);
+      ++next_verifier_index_;
+    }
```
Comment (fgerlits) [inline extensions/http-curl/tests/C2MultipleCommandsTest.cpp:58]: nitpicking, but I would prefer this:
```suggestion
    ack_verifiers_[next_verifier_index_](root);

    ++next_verifier_index_;
    if (next_verifier_index_ >= ack_verifiers_.size()) {
      next_verifier_index_ = 0;
    }
```

Comment (lordgamez) [reply extensions/http-curl/tests/C2MultipleCommandsTest.cpp]: Updated in 7752fe67bfe1ede48c5c9e8e76b50868a39bd139

**Review by fgerlits — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**